test/cpp/jit/tests.h

```diff
@@ -2067,6 +2068,17 @@ void testAliasAnalysis() {
         usesB->node(), mutatesAliasOfB->node()));
   }
 }
+
+void testEvalModeForLoadedModule() {
+  std::string module_path = "dropout_model.pt";
+  std::shared_ptr<torch::jit::script::Module> module = torch::jit::load(module_path);
```

Comment:
Maybe first assert that we start out in training mode `JIT_ASSERT(module->get_module("dropout")->is_training());`